impala/sqlalchemy.py

```diff
@@ -182,9 +182,27 @@ def get_foreign_keys(self, connection, table_name, schema=None, **kw):
         return []
 
     def get_indexes(self, connection, table_name, schema=None, **kw):
-        # no indexes in impala
-        # TODO(laserson): handle partitions, like in PyHive
-        return []
+        name = table_name
+        if schema is not None:
+            name = '%s.%s' % (schema, name)
+        query = 'DESCRIBE FORMATTED %s' % name
+        cursor = connection.execute(query)
+        rows = cursor.fetchall()
+        # Strip whitespace
+        rows = [[col.strip() if col else None for col in row] for row in rows]
+        # Filter out empty rows and comment
+        rows = [row for row in rows if row[0]]
+        for i, (col_name, _col_type, _comment) in enumerate(rows):
+            if col_name == '# Partition Information':
+                break
+        # Handle partition columns
+        col_names = []
+        for col_name, _col_type, _comment in rows[i + 1:]:
```

Comment:
How does this work? It seems like this is going to process all of the "DESCRIBE FORMATTED" output after "#Partition Information" regardless of whether it's a comment or not.

Reply:
My best guess is # Partition Information was last in describe output at that time in impala. Will add a fix in the next few days when I catch some time.